### PR TITLE
Version updates

### PR DESCRIPTION
**Pull request by dnlkoch**
## Description

This updates all (dev-) dependencies to their latest versions. Noteable updates:

- TypeScript has been updated to version 5 (including some minor fixes for suppressed warnings).
- Antd has been update to version 5 (including the removal of deprecated component usages).
- Removal of several unused dependencies (e.g. babel and moment).
- No audit warnings left.

Please note that this should be addressed as breaking since the peers of `antd` and `@ant-design/icons` require version 5 now. Refer to the official [migration guide](https://ant.design/docs/react/migration-v5) for details.

Please review.

## Do you introduce a breaking change?

- [x] Yes
- [ ] No
- [ ] I am unsure (no worries, we'll find out)

### CODE REVIEW

File: src/Util/UnsupportedPropertiesUtil.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ class UnsupportedPropertiesUtil {
     const unsupportedSymbolizerProps = UnsupportedPropertiesUtil
       .getUnsupportedPropsForSymbolizer<T>(unsupportedProperties, symbolizerName);
 
-    const val: SupportDef = unsupportedSymbolizerProps?.[(propName as string)];
+    const val = unsupportedSymbolizerProps?.[(propName as keyof UnsupportedSymbolizerProps<T>)] as SupportDef;
```
Comment (KaiVolland) [inline src/Util/UnsupportedPropertiesUtil.ts:68]: Can you extract `propName as keyof UnsupportedSymbolizerProps<T>` to a seperate type to enhance readability.

File: src/Util/UnsupportedPropertiesUtil.ts [hunk 2]
```diff
@@ -91,7 +91,7 @@ class UnsupportedPropertiesUtil {
     unsupportedProperties: UnsupportedProperties,
     symbolizerName: SymbolizerName
   ): UnsupportedSymbolizerProps<T> => {
-    return unsupportedProperties?.Symbolizer?.[symbolizerName];
+    return unsupportedProperties?.Symbolizer?.[symbolizerName as keyof UnsupportedProperties['Symbolizer']];
```
Comment (KaiVolland) [inline src/Util/UnsupportedPropertiesUtil.ts:94]: `symbolizerName as keyof UnsupportedProperties['Symbolizer']` --> see above

**Review by KaiVolland — APPROVED** [review]
Very nice. Just two minor remarks regarding readability.

File: src/Util/UnsupportedPropertiesUtil.ts (hunk 1 above)
Comment (dnlkoch) [reply src/Util/UnsupportedPropertiesUtil.ts]: See ca81f698.

File: src/Util/UnsupportedPropertiesUtil.ts (hunk 2 above)
Comment (dnlkoch) [reply src/Util/UnsupportedPropertiesUtil.ts]: See ca81f698.

**Comment by dnlkoch** [comment]
Thanks for the review!

**Pull request merged**